src/controllers/docs.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+import redoc from 'redoc-express';
+
+import { Controller } from '../interfaces';
+
+export class DocsController implements Controller {
+
+  public basepath = '/docs';
+
+  public boot(): Router {
+    const router = Router();
+    
+    router.get('/docs/openapi.yaml', (req, res) => {
+      res.sendFile('.../openapi.yaml');
+    });
+
+    router.post(
+      `${this.basepath}`,
+      redoc({
+        title: 'API Docs',
+        specUrl: 'docs/openapi.yaml'
+      })
+    );
+
```

Comment:
this should change a bit 🙂. First we should use the `GET` method, then we should fix the `specUrl` so that it finds the right endpoint. Since this is a controller, it has the version prefix `v1` set in here:
https://github.com/asyncapi/server-api/blob/ef567920029d343c735c2d980cca23c9d2804c31/src/app.ts#L66

How about we change to something like this:
```suggestion
    router.get(
      this.basepath,
      redoc({
        title: 'API Docs',
        specUrl: `/v1${this.basepath}/openapi.yaml`
      })
    );
```
You could also create a constant variable (maybe in `app.ts`) and export it to use in this controller. The string "v1" should only be in one place 🙂. 

Or we could simply not have a controller for this, and just be a function on `app.ts` like how other middlewares are set up. Then we wouldn't have the version prefix, plus the work of changing it when we have new versions.

Reply:
in app.ts (new lines of code)

```
import redoc from 'redoc-express';
 private initializeMiddlewares() {
   .....
   ......
    this.app.use(redoc({
        title: 'API Docs',
        specUrl: `../openapi.yaml`
      }));
  }
```
However how could I specify the base path `/docs`?

Reply:
@everly-gif  I would leave the controller and go with the exported value `v1` solution. The `app.ts` is the place for generic middlewares that are used for all endpoints and not for individual ones :) Please propagate suggestions from @BOLT04 (change `post` to the `get`) and should be fine :)

Reply:
yeah I think it looks good @magicmatatjahu  👍